guppylang/tys/ty.py

```diff
@@ -376,12 +390,13 @@ def intrinsically_linear(self) -> bool:
 
     def to_hugr(self) -> tys.Type:
         """Computes the Hugr representation of the type."""
-        if all(
-            isinstance(e, TupleType) and len(e.element_types) == 0
-            for e in self.element_types
-        ):
-            return tys.UnitSum(size=len(self.element_types))
-        return tys.GeneralSum(row=[t.to_hugr() for t in self.element_types])
+        rows = [type_to_row(ty) for ty in self.element_types]
+        if all(len(row) == 0 for row in rows):
```

Comment:
Consider
```
sum_inner = tys.UnitSum(size=len(rows)) if all(len(row)==0 for row in rows) else tys.GeneralSum(rows=[[ty.to_hugr() for ty in row] for row in rows])
return tys.Type(tys.TaggedSumType(st=tys.SumType(sum_inner)))
```

Also the `[[ty.to_hugr() for ty in row] for row in rows]` could be a useful utility method, there are quite a few such double-traversals. `sum_variants_to_hugr`, say